discovery-provider/src/queries/get_track_library.py

```diff
@@ -86,154 +85,122 @@ def _get_track_library(args: GetTrackLibraryArgs, session):
     # union them together and remove duplicate track IDs.
     #
     # It's a straight forward approach with a few SQLALchemy gotchas - namely
-    # that in the case of querying 'all', we need to alias the TrackWithAggregates table
-    # to avoid accidentally joining on TrackWithAggregates, and in subsequent query building logic
-    # conditionally refer to either the aliased TrackWithAggregatesAlias table or the raw TrackWithAggregates table
-    # to avoid an accidental join.
-
-    favorites_base = (
-        session.query(TrackWithAggregates, Save.created_at.label("item_created_at"))
-        .join(Save, Save.save_item_id == TrackWithAggregates.track_id)
-        .filter(
-            TrackWithAggregates.is_current == True,
-            Save.user_id == user_id,
-            Save.is_current == True,
-            Save.is_delete == False,
-            Save.save_type == SaveType.track,
-        )
+    # we have to use contains_eager when including the user for the query filter so that it doesn't get joined twice
+
+    favorites_base = session.query(
+        Save.save_item_id.label("item_id"), Save.created_at.label("item_created_at")
+    ).filter(
+        Save.user_id == user_id,
+        Save.is_current == True,
```

Comment:
Unrelated to your change I feel like `is_current` and `is_delete` queries should have automatic values for the most common case. It feels error-prone that you have to remember to specific both of these when implementing a new filter in order to get the most common expected behavior 😬